userspace/falco/app_actions/configure_interesting_sets.cpp

```diff
@@ -15,27 +15,106 @@ limitations under the License.
+	std::stringstream ss;
+	std::copy(names.begin(), names.end(),
+	std::ostream_iterator<std::string>(ss, ", "));
+	std::string names_str = ss.str();
+	return names_str.substr(0, names_str.size() - 2);
+}
 
-	/* Please note: here we fill these 2 sets because we are interested in only some features, if we leave
-	 * them empty `libsinsp` will fill them with all the available syscalls and all the available tracepoints!
-	 */
+void application::extract_rules_event_names(std::unique_ptr<sinsp>& inspector, std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* Get all PPME events from all rules as idx codes. */
+	std::set<uint16_t> rule_events;
+	std::string source = falco_common::syscall_source;
+	m_state->engine->evttypes_for_ruleset(source, rule_events);
+	std::unordered_set<uint32_t> ppme_events_codes(rule_events.begin(), rule_events.end());
+
+	/* Translate PPME event codes to event names. */
+	rules_evttypes_names = inspector->get_events_names(ppme_events_codes);
+}
 
-	/* Here the `libsinsp` state set is not enough, we need other syscalls used in the rules,
-	 * so we use the `simple_set`, this `simple_set` contains all the syscalls of the `libsinsp` state
-	 * plus syscalls for Falco default rules.
+void application::activate_interesting_syscalls_events(std::unique_ptr<sinsp>& inspector, const std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* PART 1: syscalls and events activation
+	 *
+	 * Current enforce_simple_ppm_sc_set approach includes 2 steps:
+	 * (1) Enforce all syscalls from all Falco rules
+	 * (2) Enforce a static set of syscalls that includes all syscalls for the Falco default rules
+	 * (3) Enforce `libsinsp` state set (non-adaptive, not conditioned by rules, but based on PPME event table flags indicating generic sinsp state modifications)
+	 * -> Final set is union of (1), (2) and (3)
+	 *
+	 * New user input ONLY approach via m_state->config->m_additional_syscalls option:
+	 * Additional syscalls of interest providen by the end user in addition to the syscalls/events from all rules. Defaults to EMPTY and
+	 * sinsp state enforcement is applied instead when this option is not set
+	 *
+	 * TODO:
+	 * - Support multiple new more adaptive syscalls activation options as alternatives via new Falco flags to provide maximum control and configurability to end users
 	 */
-	m_state->ppm_sc_of_interest = inspector->enforce_simple_ppm_sc_set();
+
+	/* Translate PPME event names to PPM syscall codes. */
+	std::unordered_set<uint32_t> ppm_sc_set = {};
+	for (int ppm_sc_code = 0; ppm_sc_code < PPM_SC_MAX; ++ppm_sc_code)
+	{
+		std::string ppm_sc_name = g_infotables.m_syscall_info_table[ppm_sc_code].name;
+		if (rules_evttypes_names.find(ppm_sc_name) != rules_evttypes_names.end())
+		{
+			ppm_sc_set.insert(ppm_sc_code);
+		}
+	}
+
+	falco_logger::log(LOG_INFO, "(" + std::to_string(rules_evttypes_names.size()) + ") event types activated in rules: " + concat_evttypes_names(rules_evttypes_names) + "\n");
```

Comment:
Here we might use `inspector->get_syscalls_names(ppm_sc_set);`

Reply:
Done, alongside more cleanup around comments and naming to further increase technical clarity.